pkg/loop/config.go

```diff
@@ -0,0 +1,218 @@
+	return c.contractTracker
+}
+
+var _ libocr.OffchainConfigDigester = (*offchainConfigDigesterClient)(nil)
+
+type offchainConfigDigesterClient struct {
+	grpc pb.OffchainConfigDigesterClient
+}
+
+func (o *offchainConfigDigesterClient) ConfigDigest(config libocr.ContractConfig) (digest libocr.ConfigDigest, err error) {
+	pbCfg := pbContractConfig(config)
+	var reply *pb.ConfigDigestReply
+	reply, err = o.grpc.ConfigDigest(context.TODO(), &pb.ConfigDigestRequest{
+		ContractConfig: pbCfg,
+	})
+	if err != nil {
+		return
+	}
+	if l := len(reply.ConfigDigest); l != 32 {
+		err = ErrConfigDigestLen(l)
+		return
+	}
+	copy(digest[:], reply.ConfigDigest)
+	return
+}
+
+func (o *offchainConfigDigesterClient) ConfigDigestPrefix() libocr.ConfigDigestPrefix {
+	reply, err := o.grpc.ConfigDigestPrefix(context.TODO(), &pb.ConfigDigestPrefixRequest{})
+	if err != nil {
+		panic(err) //TODO retry https://smartcontract-it.atlassian.net/browse/BCF-2112
+	}
+	return libocr.ConfigDigestPrefix(reply.ConfigDigestPrefix)
+}
+
+var _ pb.OffchainConfigDigesterServer = (*offchainConfigDigesterServer)(nil)
+
+type offchainConfigDigesterServer struct {
+	pb.UnimplementedOffchainConfigDigesterServer
+	impl libocr.OffchainConfigDigester
+}
+
+func (o *offchainConfigDigesterServer) ConfigDigest(ctx context.Context, request *pb.ConfigDigestRequest) (*pb.ConfigDigestReply, error) {
+	if request.ContractConfig.F > math.MaxUint8 {
+		return nil, ErrUint8Bounds{Name: "F", U: request.ContractConfig.F}
+	}
+	cc := libocr.ContractConfig{
+		ConfigCount:           request.ContractConfig.ConfigCount,
+		F:                     uint8(request.ContractConfig.F),
+		OnchainConfig:         request.ContractConfig.OnchainConfig,
+		OffchainConfigVersion: request.ContractConfig.OffchainConfigVersion,
+		OffchainConfig:        request.ContractConfig.OffchainConfig,
+	}
+	copy(cc.ConfigDigest[:], request.ContractConfig.ConfigDigest)
+	for _, s := range request.ContractConfig.Signers {
+		cc.Signers = append(cc.Signers, s)
+	}
+	for _, t := range request.ContractConfig.Transmitters {
+		cc.Transmitters = append(cc.Transmitters, libocr.Account(t))
+	}
+	cd, err := o.impl.ConfigDigest(cc)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.ConfigDigestReply{ConfigDigest: cd[:]}, nil
+}
+
+func (o *offchainConfigDigesterServer) ConfigDigestPrefix(ctx context.Context, request *pb.ConfigDigestPrefixRequest) (*pb.ConfigDigestPrefixReply, error) {
+	p := o.impl.ConfigDigestPrefix()
+	return &pb.ConfigDigestPrefixReply{ConfigDigestPrefix: uint32(p)}, nil
+}
+
+var _ libocr.ContractConfigTracker = (*contractConfigTrackerClient)(nil)
+
+type contractConfigTrackerClient struct {
+	grpc pb.ContractConfigTrackerClient
+}
+
+func (c *contractConfigTrackerClient) Notify() <-chan struct{} { return nil }
+
+func (c *contractConfigTrackerClient) LatestConfigDetails(ctx context.Context) (changedInBlock uint64, configDigest libocr.ConfigDigest, err error) {
```

Comment:
In most of these cases it was just more succinct to express this way, rather than e.g. add a line to declare each of these and a bunch of extra characters on the returns since we have value types with long names.